controllers/storageclass/storageclass_controller.go

```diff
@@ -195,6 +196,53 @@ func (r *StorageClassWatcher) getFlashSystemClusterByStorageClass(sc *storagev1.
 	return foundCluster, fmt.Errorf(msg)
 }
 
+func (r *StorageClassWatcher) getManagementMapFromSecret(topologySecret corev1.Secret) (map[string]v1alpha1.FlashSystemCluster, error) {
+	managementMap := make(map[string]v1alpha1.FlashSystemCluster)
+	jsonSecretConfigData := make(map[string]interface{})
+	var topologySecretData []byte
+	//topologySecretData = topologySecret.Data["config"]
+	for _, value := range topologySecret.Data {
+		topologySecretData = value
+		break
+	}
+	// decoding topologySecretData from base64 to json
+	decodedSecretData, err := base64.StdEncoding.DecodeString(string(topologySecretData))
+	if err != nil {
+		r.Log.Error(nil, "failed to decode configData from topology secret")
+		return managementMap, err
+	}
+	if err = json.Unmarshal(decodedSecretData, &jsonSecretConfigData); err != nil {
+		r.Log.Error(nil, "failed to unmarshal configData")
+		return managementMap, err
+	}
+	clusters := &v1alpha1.FlashSystemClusterList{}
+	if err = r.Client.List(context.Background(), clusters); err != nil {
+		r.Log.Error(nil, "failed to list FlashSystemClusterList")
+		return managementMap, err
+	}
+	for mgmt_id, mgmt_data := range jsonSecretConfigData {
+		mgmt_address := mgmt_data.(map[string]interface{})["management_address"]
+		for _, cluster := range clusters.Items {
+			clusterSecret := &corev1.Secret{}
+			err = r.Client.Get(context.Background(),
+				types.NamespacedName{
+					Namespace: cluster.Spec.Secret.Namespace,
+					Name:      cluster.Spec.Secret.Name},
+				clusterSecret)
```

Comment:
I think it will be better to retrieve the secret from the cluster only once, before the jsonSecretConfigData loop, and save it locally, instead of reading the object from the cluster for every mgmt_id.